### PR TITLE
#11920 move mypy config to pyproject.toml

### LINKED ISSUE #11920: move mypy config to pyproject.toml
**Is your feature request related to a problem? Please describe.**
mypy config is currently in the old .ini, mypy now has support for pyproject.toml

**Describe the solution you'd like**
move mypy.ini config to pyproject.toml

### PR DESCRIPTION
**Pull request by graingert**
## Scope and purpose

Fixes #11920

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -288,3 +288,383 @@ known_zope = "zope"
...
+    'twisted.python.test.test_win32',
+    'twisted.python.test.test_zippath',
+    'twisted.python.threadpool',
+    'twisted.python.zippath',
+    'twisted.runner.inetdtap',
+    'twisted.runner.procmontap',
+    'twisted.runner.inetdconf',
+    'twisted.runner.test.test_procmontap',
+]
+
```
Comment (graingert) [inline pyproject.toml:615]:
```suggestion
```

pretty sure this config block is redundant

Comment (adiroiban) [reply pyproject.toml:615]: Feel free to delete this.
As long as mypy check is green, this should be ok.

Thanks

**Review by adiroiban — APPROVED** [review]
Thanks for the cleanup!

Please merge as  you want :)

File: pyproject.toml (hunk 1 above)
Comment (graingert) [reply pyproject.toml:615]: I'll do this in a follow up I think

**Pull request merged**